### ISSUE TITLE
graph_view_un_el.cpp

### ISSUE THREAD

**Issue by kuvshinovdr** [kuvshinovdr/OGxx#34]
1. Обновить свою ветку по main.
2. Создать файл source/graph_view_un_el.cpp
3. В нём определить класс ogxx::Graph_view_undirected_edge_list: public ogxx::Graph_view.
4. В данный класс в качестве поля добавить ссылку на Edge_list (инициализировать конструктором).
5. Реализовать методы Graph_view поверх Edge_list, предполагая, что это неориентированный граф. Считать, что пары вершин, представляющие рёбра, должны быть отсортированы, т.е. не (2, 0), а (0, 2) (поскольку в неориентированном графе (0, 2) и (2, 0) представляют одно и то же ребро).
6. В том же файле реализовать функцию ogxx::undirected::graph_view(Edge_list& el), которая будет возвращать новый объект ogxx::Graph_view_undirected_edge_list, инициализированный ссылкой el.

**Comment by kuvshinovdr** [comment]
Добавлен source/graph_view_di_al.cpp, реализующий directed::graph_view для Adjacency_list.

**Comment by kuvshinovdr** [comment]
Максим, в ветке не видно прогресса -- не выполнен даже пункт 2. Задание было включено в milestone, истёкший две недели назад.

**Comment by Pinkgun89** [comment]
Здравствуйте, реализовал graph_view_un_el

**Comment by Pinkgun89** [comment]
Здравствуйте, я сделал graph_veiw_un_el.

сб, 2 дек. 2023 г. в 01:40, Dmitry R. Kuvshinov ***@***.***>:

**Comment by kuvshinovdr** [comment]
Я сделал коммит в вашу ветку с дополнениями и исправлениями. В вашем .cpp-файле поставил TODO в некоторых функциях. Посмотрите, что в нём изменено, и попробуйте исправить код в тех местах, где поставлено TODO.

**Comment by Pinkgun89** [comment]
Здравствуйте, Дмитрий Рустамович. Для исправления мне самому нужно
реализовать sorted_erase и sorted_insert?

чт, 28 дек. 2023 г. в 14:10, Dmitry R. Kuvshinov ***@***.***>:

**Comment by kuvshinovdr** [comment]
Нет, нужно реализовать без них через методы St_set\<Vertex_pair\>, если это доступно, или через методы Edge_list, если недоступно.
